packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart

```diff
@@ -25,7 +25,16 @@ class FirestoreDepositAdapter extends DepositAdapter<String> {
     String primaryColumn,
     Map<String, dynamic> data,
   ) async {
-    final value = await _firestore.collection(table).add(data);
+    assert(
+      data[primaryColumn] == null,
+      'Firestore adapter only works with auto generated ids, '
```

Comment:
We remove it anyway, so does this check need to exist? I was also thinking of always removing the primaryColumn on adding in supabase adapter

Reply:
I think it would be good, because otherwise the user may be intentionally try to pass an id, that will not be used and be confused

Reply:
The problem is that entities have a `toJSON` method, and you would then map that data. I think it is the adapter job to handle this without developers having to think about that.

For instance I now constantly have to write:
```dart
class TestEntity extends Entity {
  int? id;

  Map<String, dynamic> toJSON() {
    return {
      if (id != null) 'id': id,
      // ..etc
    };
  }
}
```

Reply:
Sounds good, removed!